addons/mail/static/src/core/common/action.js

```diff
@@ -116,7 +117,10 @@ export class Action {
 
     /** Condition to display the action panel component of this action. */
     get actionPanelComponentCondition() {
-        return this.isActive && this.actionPanelComponent && this.condition && !this.popover;
+        return (
+            this.definition.actionPanelComponentCondition?.call(this, this.params) ??
+            (this.isActive && this.actionPanelComponent && this.condition && !this.popover)
+        );
```

Comment:
Can use `(typeof val === "function" ? val.call(this, params) : val) ?? (isActive && ...)` so that primitive values in explicit definition are allowed, e.g. `actionPanelComponentCondition: true`.